src/App.js

```diff
@@ -47,22 +66,48 @@ const App = () => {
     //    3 squares in each column match
     // 3. Go across each diagonal to see if 
     //    all three squares have the same value.
+    
+    //check if winner is in rows else they are in columns:
+    for (let i = 0; i < 3; i++) {
+      if (squares[i][0].value === squares[i][1].value && squares[i][1].value === squares[i][2].value && squares[i][0].value !== '') {
+        return squares[i][0].value;
+      } else if (squares[0][i].value === squares[1][i].value && squares[1][i].value === squares[2][i].value && squares [0][i].value !== '') {
+        return squares[0][i].value;
+      } 
+    }
+    //winner in both diagonals:
+    if (squares[0][0].value === squares[1][1].value && squares[1][1].value === squares[2][2].value & squares[0][0].value !== '') {
+      return squares[0][0].value;
+    } else if (squares[0][2].value === squares[1][1].value && squares[1][1].value === squares[2][0].value && squares[0][0].value !== '') {
+      return squares[0][2].value;
+    } 
 
-  }
+    for (let r = 0; r < 3; r++) {
+      for (let c = 0; c < 3; c++) {
+        if (squares[r][c].value === '') {
+          return null;
+        }
+      }}
+      //returning null if there are empty squares
+      return 'Tie';
+
+  };
 
-  const resetGame = () => {
+  function reset() {
     // Complete in Wave 4
+    setSquares(generateSquares())
   }
 
   return (
     <div className="App">
       <header className="App-header">
         <h1>React Tic Tac Toe</h1>
-        <h2>The winner is ... -- Fill in for wave 3 </h2>
-        <button>Reset Game</button>
+        <h2>Current player is {currentPlayer}</h2>
+        <h2>The winner is ...{winner} </h2>
```

Comment:
This works, but it doesn't quite pass the tests.  The tests are looking for a very specific text ("Winner is x" or "Winner is o") so adding the following changes will result in all the required tests passing.
```suggestion
        <h2>Winner is {winner} </h2>
```

This does not display the "..." in the example version though.  One way to handle this is to create a function, for example:
```
const getWinnerFeedback = () => {
    if(winner != null && winner != "Tie") {
        return "Winner is " +  winner;
    } else if(winner == "Tie") {
        return "It's a tie!"
     } else {
        return "The winner is ..."
}
```
Then on line 106 call the function:
```suggestion
        <h2>getWinnerFeedback()</h2>
```